core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -1804,6 +1798,21 @@ class WorkspaceService(protected val userInfo: UserInfo, val dataSource: SlickDa
     }
   }
 
+  def getGenomicsOperation(workspaceName: WorkspaceName, jobId: String): Future[PerRequestMessage] = {
+    // First check the workspace context and permissions in a DB transaction.
+    // We don't need any DB information beyond that, so just return Unit on success.
+    dataSource.inTransaction { dataAccess =>
+      withWorkspaceContextAndPermissions(workspaceName, WorkspaceAccessLevels.Read, dataAccess) { _ =>
+        DBIO.successful(())
+      }
+    }
+    // Next call GenomicsService, which actually makes the Google Genomics API call.
+    .flatMap { _ =>
```

Comment:
Actually `andThen` wouldn't work in this case because it returns the result of the _first_ future. Here I want to return the result of the second future, but only if the first future is successful. So I think `flatMap` is appropriate.